### PR TITLE
Wiki redesign 

### PR DESCRIPTION
**Pull request by C-Loftus**
Addresses #235 

# Draft Images / Layout
image
image

Would appreciate any feedback on this as I work on refactoring. Will cc community stakeholders before merging / finalizing.

This is easy to run locally just run

`yarn; yarn start` Works with npm too 

@2shea / @splondike any interim feedback?

Todo items: 
- [x] Set up docusaurus
- [x] Align on CSS formatting
- [ ] Set up Github Pages Deployment / transfer netlify to new deployment
- [ ] Get Feedback and Align on Homepage Formatting / Any Images
- [x] Get higher Res/Transparent Talon logo
- [ ] Get rid of any legacy Github actions / Netlify integrations that are no longer needed
- [ ] Change Github repo to have proper description, not "Wiki for Talon Beta"

### CODE REVIEW

**Comment by C-Loftus** [comment]
@2shea I did a lot of work with this. It is very close to being ready. When you get a chance, perhaps we can briefly call to discuss moving over the deployment to Netlify and aligning on what the front page should look like (I think the old welcome page is too wordy

The search is easy to add and is from : https://docsearch.algolia.com/ This is free but requires us to apply and then put a code snippet in. It is free and will not be an issue, we just need to have the site live with docusaurus before adding it.

**Comment by C-Loftus** [comment]
Yes sounds good! Please feel free to ping me on slack and we can find a time this week to chat

**Comment by saidelike** [comment]
Happy to be part of the call or to help by reviewing what you've done.

**Comment by C-Loftus** [comment]
@splondike I have been again thinking about formatters. Although I know that you mentioned previously that introducing a formatter now messes up the git blame, I think it is worth discussing again since things will be changing a lot and a good amount of the wiki will probably be checked over in the coming month or two so the git blame might not be as relevant for this specific commit. 

I originally thought that maybe we could have an editor by default but I'm realizing as well but it is hard to enforce this if people are editing outside VSCode where we can't control editor settings.  So essentially what I did in the most recent commit is just disable format on save in visual studio code workspace settings.

Just wanted to align since  if there is ever a time to change it it would be now.

**Comment by splondike** [comment]
New site's looking good, nice job!

Regarding formatting there's three approaches I've seen:

1. Don't do it automatically, write a style guide somewhere potentially.
2. Run a check in CI that fails the PR if it doesn't conform.
3. Run something in CI that autofixes the styling and commits that to the branch on their behalf when somebody makes a PR. Something like [this](https://peterevans.dev/posts/github-actions-how-to-automate-code-formatting-in-pull-requests/).

I'm not sure option 2 would be a great fit for this project, we probably don't want to make it any harder for people to contribute. So that leaves the other two. Personally I don't really like autoformatters, even in programming languages; the cases where they make things worse always annoy me. I'm happy to use them if other people do though.

What do you reckon?

**Comment by C-Loftus** [comment]
> Regarding formatting there's three approaches I've seen...
 So I think the first option will probably be ignored and I agree with the second option being a bit too strict.  I think num 3, using prettier in a pre-commit hook works. It would still potentially mess up the git blame though unless we have another pre commit hook to git blame ignore those specific commits. I will get pokey's  opinion on this perhaps during Sunday's cursorless meet up since he is the CI / devtool wizard :) and helped me setup precommit in talon-ai-tools. 

Only other thing that is a blocker is visual styling.  I was told aegis  wants to restrict the talon logo to official work and not community repos.  Which is fair enough, but not sure what to put for the website logo/favicon/front page graphics.  The previous wiki didn't have any visual styling and even though it sounds like fluff,  I think it is quite important to have appealing visual design to hook people in.

**Comment by splondike** [comment]
Hopefully the autoformatter wouldn't be conflicting with what people do too much, so I wouldn't worry about messing with the git blame in this case. You can reformat the entire codebase as one commit and people can ignore that. Subsequent modifications with their own local reformattings are easier to manage then.

I don't much mind about logos, happy to see what people come up with :).

**Comment by C-Loftus** [comment]
Talked with emily:

Refactor for the time being will be hosted at: https://talon-wiki-refactor.netlify.app/ which will point at a staging branch
Rebase the commit into the `staging` branch or some other permanent branch that we can use for the staging preview

**Comment by C-Loftus** [comment]
Merging this into the staging branch. Still not merged into main. We needed to merge into the staging branch given the fact the netlifly deployment deploys from staging

**Review by 2shea — APPROVED** [review]
Looks good to merge to staging! I will configure a new site in netlify to deploy from the staging branch.

**Comment by saidelike** [comment]
Looks really nice.

It might be normal but:
Edit this page on https://talon-wiki-refactor.netlify.app/Talon-Intro/basic_usage brings to 404.

**Comment by saidelike** [comment]
Maybe re-order sections.
https://talon-wiki-refactor.netlify.app/Talon-Intro/basic_usage

For instance, what about 
1) getting started
2) basic usage

https://talon-wiki-refactor.netlify.app/Talon-Scripting/unofficial_talon_docs what about renaming this part "talon programming". All the community wiki is unofficial docs, not just this part

**Comment by C-Loftus** [comment]
Yup, we agreed that those should be renamed. 404 is also no longer an issue. 

Please discuss all new comments in the new PR https://github.com/TalonCommunity/Wiki/pull/237 , sorry we had to so some git shenanigans :)
